.github/actions/setup/action.yml

```diff
@@ -0,0 +1,20 @@
+name: Setup build environment
+description: Setup build environment
+runs:
+  using: "composite"
+  steps:
+    - uses: cashapp/activate-hermit@v1
+      with:
+        cache: true
+    - id: find-go-build-cache
+      shell: bash
+      run: echo "cache=$(go env GOCACHE)" >> $GITHUB_OUTPUT
+    - uses: actions/cache/restore@v4
```

Comment:
You'll also need the `writecache.yml` action

Reply:
Ya know, I merged this without realizing this was causing a failure... latest on main is :x: https://github.com/TBD54566975/web5-go/actions/runs/7920575975/job/21623911141

@alecthomas that `writecache.yml` is relevant to the build stage, but this repo doesn't have a build stage in our Justfile... should we?

Reply:
https://github.com/TBD54566975/web5-go/pull/55